lib/routing/sfRoute.class.php

```diff
@@ -279,6 +270,45 @@ public function generate($params, $context = [], $absolute = false)
         return $url;
     }
 
+    /**
+     * Generates a consistent cache key for a route
+     */
+    protected function generateCacheKey(array $elements = [], bool $absolute = false): string
+    {
+        $elements = [
+            SharedCacheHelper::ROUTING_NAMESPACE,
+            $this->flattenArrayElements($elements),
+        ];
+
+        if ($absolute) {
+            $elements[] = 'Absolute';
+        }
+
+        return SharedCacheHelper::getNamespace($elements);
+    }
+
+    /**
+     * Recursive method to pack a key => value pair array down into a flat
+     * numerically-indexed array, with a consistent ordering of elements.
+     */
+    protected function flattenArrayElements(array $elements = []): array
```

Comment:
Interesting. Should this method be in some sort of helper class?

Reply:
Umm....it's a good question, but I don't know if there are any other use cases for that functionality to be shared with.

I would definitely like to see the functionality shifted to a more centralised location if we have need for sharing it, but I would generally stick things close to where they're needed until we have a good reason to move it really.